be/src/olap/memtable.cpp

```diff
@@ -263,22 +274,56 @@ vectorized::Block MemTable::_collect_vskiplist_results() {
                 auto function = _agg_functions[i];
                 function->insert_result_into(it.key()->_agg_places[i],
                                              *(_output_mutable_block.get_column_by_position(i)));
-                function->destroy(it.key()->_agg_places[i]);
+                if (final) {
+                    function->destroy(it.key()->_agg_places[i]);
+                }
             }
+            // re-index the row_pos in VSkipList
+            it.key()->_row_pos = idx;
+            idx++;
+        }
+        if (!final) {
+            _input_mutable_block.swap(_output_mutable_block);
+            //TODO(weixang):opt here.
+            _output_mutable_block = vectorized::MutableBlock::build_mutable_block(&in_block);
+            _output_mutable_block.clear_column_data();
+        }
+    }
+}
+
+void MemTable::shrink_memtable_by_agg() {
+    {
+        SCOPED_TIMER(_shrunk_agg_time);
+        if (_is_shrunk_by_agg) {
+            return;
         }
+        size_t old_size = _input_mutable_block.allocated_bytes();
+        _collect_vskiplist_to_output(false);
+        size_t new_size = _input_mutable_block.allocated_bytes();
+        // shrink mem usage of memetable after agged.
+        _mem_usage += new_size - old_size;
+        _mem_tracker->consume(new_size - old_size);
+        _is_shrunk_by_agg = true;
     }
-    return _output_mutable_block.to_block();
+}
+
+bool MemTable::is_full() {
+    return memory_usage() >= config::write_buffer_size;
 }
 
 Status MemTable::flush() {
-    VLOG_CRITICAL << "begin to flush memtable for tablet: " << _tablet_id
+    clock_t now = clock();
```

Comment:
do too many log operation

Reply:
i pushed my debug code by mistake.
These logs and profiles will be removed later.